### PR TITLE
feat: add LokiExchangeStore for sending exchanges to Grafana Loki

### PR DESCRIPTION
**Pull request by rrayst**
Introduced a new exchange store implementation, `LokiExchangeStore`, that pushes completed and failed exchanges to Grafana Loki for visualization using LogQL and Grafana. Includes batching, JSON serialization, and support for labels like `job` and `api`. Added unit tests for proper functionality, including error handling and edge cases.

## Summary by CodeRabbit

* **New Features**
  * Added a write-only exchange store that sends completed and failed exchange data to Loki.
  * Supports batched delivery, rule-based stream labeling, timestamps, and optional organization headers.
  * Provides configurable Loki endpoint, job, organization ID, and HTTP client settings.

* **Reliability**
  * Loki delivery failures are logged without interrupting ongoing exchange processing.

### CODE REVIEW

File: core/src/main/java/com/predic8/membrane/core/exchangestore/LokiExchangeStore.java [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    private HttpClient client;
+    private ObjectMapper mapper;
+
+    /**
+     * Exchanges seen but not yet pushed. Needed because {@link AbstractPersistentExchangeStore#snap(AbstractExchange,
+     * com.predic8.membrane.core.interceptor.Interceptor.Flow)} looks up the request-phase snapshot to merge the response
+     * into, and unlike other persistent stores we cannot read it back from Loki.
+     */
+    private final Cache<Long, AbstractExchangeSnapshot> pending = CacheBuilder.newBuilder()
+            .expireAfterWrite(5, TimeUnit.MINUTES).build();
```
[the remark this answers is not in the document]
Comment (rrayst) [reply core/src/main/java/com/predic8/membrane/core/exchangestore/LokiExchangeStore.java:81]: This is not considered to be an issue. Elsewise the design of all 'ExchangeStore' implementations writing to immutable stores has to be rethought.

File: core/src/main/java/com/predic8/membrane/core/exchangestore/LokiExchangeStore.java [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+            Response response = exc.getResponse();
+            if (response.getStatusCode() < 200 || response.getStatusCode() > 299)
+                log.error("Loki rejected {} exchanges with status {}: {}", exchanges.size(),
+                        response.getStatusCode(), response.getBodyAsStringDecoded());
+        } catch (Exception e) {
+            // Do not propagate: AbstractPersistentExchangeStore's update thread dies on an exception and would
+            // stop storing exchanges for the rest of the router's lifetime.
+            log.error("While pushing {} exchanges to Loki at {}.", exchanges.size(), url, e);
+        }
```
[the remark this answers is not in the document]
Comment (rrayst) [reply core/src/main/java/com/predic8/membrane/core/exchangestore/LokiExchangeStore.java:122]: Delivery is best-effort by design and consistent with the other persistent stores.

**Review by t-burch — APPROVED** [review]
